src/west/manifest.py

```diff
@@ -214,15 +214,23 @@ def _manifest_content_at(project: 'Project', path: PathType,
 
 class _import_map(NamedTuple):
     file: str
+    name_includelist: List[str]
+    '''@deprecated use name_includelist'''
     name_whitelist: List[str]
+    path_includelist: List[str]
+    '''@deprecated use path_includelist'''
     path_whitelist: List[str]
+    name_excludelist: List[str]
+    '''@deprecated use name_excludelist'''
     name_blacklist: List[str]
+    path_excludelist: List[str]
+    '''@deprecated use path_excludelist'''
```

Comment:
There's no need for this tuple to contain these "deprecated" items. Instead just have the following fields:

- file
- name_allowlist
- path_allowlist
- name_denylist
- path_denylist
- path_prefix

(I believe @tejlmand is still asking for that change and has agreed to move to those names, but correct me if I'm wrong.)

We can then populate the tuple from the parsed YAML object as follows:

```py
# Here we fall back on old names for these fields which we've since
# renamed. The old names must still work to keep manifests written
# against west 0.7 and 0.8. Those are still in people's git histories
# and we need to keep 'west update' working when those revisions are
# checked out.
name_allowlist = copy.pop('name-allowlist', copy.pop('name-whitelist', []))
path_allowlist = copy.pop('path-allowlist', copy.pop('path-whitelist', []))
name_denylist = copy.pop('name-denylist', copy.pop('name-whitelist', []))
path_denylist = copy.pop('path-denylist', copy.pop('path-whitelist', []))

ret = _import_map(copy.pop('file', _WEST_YML),
                  name_allowlist, path_allowlist,
                  name_denylist, path_denylist,
                  copy.pop('path-prefix', ''))

```

The `_import_map` type is not public API as indicated by the leading `_` and the absence of any official documentation about it here: https://docs.zephyrproject.org/latest/guides/west/west-apis.html#module-west.manifest

We are therefore free to change its implementation details as needed; there is no "deprecation" necessary because it's not API. I don't think it makes sense to keep these here if we are making this change.

Reply:
Ok, I wasn't certain that _import_map is not used outside of the package. I'll remove the fallback fields.

@tejlmand [wrote](https://github.com/zephyrproject-rtos/west/pull/440#discussion_r494119991):
> I think `include`/`exclude` would be more correct.

Reply:
>  @tejlmand wrote:
>>  I think include/exclude would be more correct.

Yes, but then I responded:

>>  Honestly if we're going to bikeshed this I would rather use 'allow' and 'block' for consistency with what other major projects seem to be doing

And @tejlmand replied:

> Consistency is king. Accepted.

Reply:
Sure, changed.